lib/screens/v2/widget_instance/common/base_alert.ex

```diff
@@ -1,48 +1,29 @@
 defmodule Screens.V2.WidgetInstance.Common.BaseAlert do
   @moduledoc """
-  Common logic for structs that implement the `Screens.V2.SingleAlertWidget` protocol.
-
-  ***
-  None of the code in this module should access struct fields directly.
-
-  In order to maintain clarity on what values these functions require,
-  use only callbacks from `Screens.V2.SingleAlertWidget` to interact with inputs.
-  If new values are required, add new callbacks to the protocol and update implementing modules accordingly.
-  ***
+  Common logic for Alerts.
```

Comment:
Mmmm yes. Got me thinking: not all functions in this BaseAlert file require location context -- `informed_entities`, `effect`, `informed_subway_routes`, specifically. Should those live in `Screens.Alerts.Alert`? At the moment, it seems to mostly handle fetching and setting on alerts, and less getting (and these functions are getters). I'm torn:

1. Should these getters be moved to `Screens.Alerts.Alert`, and then the `BaseAlert` module can be renamed `LocalAlert` or something indicating it's functions with locations attached
2. Are things better organized as they are, where the fetch-and-set stuff is stored together, and post-processing stuff is stored together?

I like #2, but then what do I call this module, and where should it go?

Reply:
I think it would make "logical" sense to put the functions that only need an Alert, no location context, in `Screens.Alerts.Alert`.
But if that makes the code here way more verbose—`Alert.informed_entities(alert)` is a bit longer than `informed_entities(alert)`—feel free to leave them.

I do think this module could stand to be renamed either way, since it is like 99% about querying an Alert combined with a LocationContext.

> what do I call this module

`LocalAlert` seems good, or `LocalizedAlert` if it doesn't feel too long. `LocatedAlert`, `ScopedAlert`, `PositionedAlert`, `AlertXPlace`? :thinkies:

> where should it go

I think since it's 100% about the data, it could live right in `Screens.<ModuleName>` similar to `Screens.Alerts.Alert`, `Screens.RouteType`, stuff like that. The fact that you can pass certain `WidgetInstance` structs to its functions is just a convenient coincidence.